hatchways/team-pine
# Profile details page integration

### What this PR does:
- Hooks the profile details page to the backend.
- Profile is retrieved based on id passed in params /profile/:id
- Requests can be sent via the request form.

### Screenshots / Video:
![image](https://user-images.githubusercontent.com/34494229/150627356-ac20ac92-b085-4714-ae5c-637ffd2c930d.png)
![image](https://user-images.githubusercontent.com/34494229/150627424-f9984a79-7e9f-4d62-98f7-b247589e6cc8.png)
![image](https://user-images.githubusercontent.com/34494229/150627475-f630b433-cdf9-4425-ab7f-7d8256725fb9.png)


### Any information needed to test this feature:
- Sign up as a new user to populate the necessary fields. Some can be edited in the profile settings page, others are not implemented and must be changed directly in the database or left as empty strings.
- Make sure you have a valid Profile to retrieve, in it you will want: name, photo (url), location, description, aboutMe, isSitter, and payRate. Photo is the most optional one (as it has a placeholder), the others will just default to empty strings. isSitter must be true or the page won't load the profile at all.

- Enter in a valid date range into the request form and submit. You should see a message that it was successful. Check your server to ensure successful creation.

### Any issues with the current functionality:
- The extra images are not currently implemented in the model, thus mock images will be used until they are there and uploadable. It should be a very easy integration once it is done.
- Not all values can be set in the settings, so naturally creating and editing a profile will not show all available information (although you can still send requests).
- Natural navigation to the profile details is not yet implemented. We currently must enter the profile id manually into the url.

closes #35
